src/choices.c

```diff
@@ -83,7 +83,9 @@ merge(struct choice *front, struct choice *back)
 	c = &head;
 
 	while (front != NULL && back != NULL) {
-		if (front->score > back->score) {
+		if (front->score > back->score ||
+                (front->score == back->score &&
+                 strcmp(front->str, back->str) < 0)) {
```

Comment:
It might be faster to use strncmp(3). Or maybe that will just lead to bad results. What do you think?